src/_pytest/config/__init__.py

```diff
@@ -1305,6 +1321,72 @@ def _getconftest_pathlist(
             values.append(relroot)
         return values
 
+    def _resolve_ini_option(
+        self,
+        name: str,
+        default_value,
+        override_value: Optional[str],
+        append_values: List[str],
+        ini_type: Optional[str],
+    ):
+        value = self.inicfg.get(name, None)
+        if override_value is None and not append_values:
+            if value is None:
+                if default_value is not None:
+                    return default_value
+                if ini_type is None:
+                    return ""
+                return []
+        else:
+            if not value and default_value:
+                value = default_value
+
+            if override_value:
+                value = override_value
+            if append_values:
+                if not value:
+                    fail(
+                        "append_ini option invalid for argument '{}' since it has no value".format(
+                            name
+                        ),
+                        pytrace=False,
+                    )
+                else:
+                    value = (
+                        value + append_values
+                        if isinstance(value, list)
+                        else "{} {}".format(value, " ".join(append_values))
+                    )
+        return value
+
+    def _get_append_ini_values(self, name: str, ini_type) -> List[str]:
+        value = []
+        # append_ini is a list of "ini=value" options
+        # append all values if multiple values are set for same ini-name,
+        # e.g. -a foo=bar1 -a foo=bar2 will append 'bar1 bar2' to foo
+        for ini_config in self._append_ini:
+            try:
+                key, append_ini_value = ini_config.split("=", 1)
+            except ValueError:
+                raise UsageError(
+                    "-a/--append-ini expects option=value style (got: {!r}).".format(
+                        ini_config
+                    )
+                )
+
+            if key == name:
+                if ini_type in ["pathlist", "args"]:
+                    value.append(append_ini_value)
+                elif ini_type == "linelist":
+                    value.append("\n{}".format(append_ini_value))
+                else:
+                    self._warn_or_fail_if_strict(
+                        "append_ini option invalid for argument '{}' with type '{}'".format(
+                            name, ini_type
+                        )
+                    )
```

Comment:
```suggestion
                elif ini_type == "bool" or ini_type is None:
                    self._warn_or_fail_if_strict(
                        "append_ini option invalid for argument '{}' with type '{}'".format(
                            name, ini_type
                        )
                    )
                else:
                    assert_never(ini_type)
```

This way if a new ini type is ever added, mypy will notify us that we need to update this code, making it impossible to forget.

(Would also need to split `if ini_type in ["pathlist", "args"]:` to two `==` checks because IIRC mypy isn't smart enough to narrow the type based on `in`).